src/server/src/routes/student.ts

```diff
@@ -64,6 +66,44 @@ router.post('/update-profile', async (req: studentRequest, res: Response) => {
     }
 });
 
+/******************************************************************************
+ *          POST Request - Apply Job - /api/student/apply-job
+ ******************************************************************************/
+
+router.post('/apply-job',
+    passport.authenticate('jwt', { session: false }),
+    async (req: Request, res: Response) => {
+        //checks that caller is a student.
+        const { role, studentId } = req.user as JWTStudent;
+        if (role !== 'student') {
+            return res
+                .status(UNAUTHORIZED)
+                .json({ error: 'User is not a student' });
+        }
+        const { jobId } = req.body;
```

Comment:
I think you are forgetting to declare the type of jobId

Reply:
How do we declare the type of jobId?

Reply:
1. Make an interface:

interface jobApplicationRequest extends Request {
    body: {
        jobId: number;
    };
}

2. Use the new interface in the route:

router.post ..... (req: jobApplicationRequest  ...

Reply:
Done.